src/components/services/SortedWalletList.ts

```diff
@@ -263,30 +263,42 @@ function matchWalletList(a: WalletListItem[], b: WalletListItem[]): boolean {
 
 /**
  * Filters a wallet list using a search string.
+ * Supports multi-word search where each word must match at least one field.
  */
 export function searchWalletList(
   list: WalletListItem[],
   searchText: string
 ): WalletListItem[] {
   if (searchText === '') return list
 
-  const target = normalizeForSearch(searchText)
+  // Split search text into individual terms (space-delimited), then normalize
+  const searchTerms = searchText
+    .split(/\s+/)
+    .filter(term => term.length > 0)
+
+  if (searchTerms.length === 0) return list
+
   return list.filter(item => {
     // Eliminate loading wallets in search mode:
     if (item.type !== 'asset') return false
     const { token, wallet } = item
 
     // Grab wallet and token information:
     const { currencyCode, displayName } = token ?? wallet.currencyInfo
+    const { chainDisplayName } = wallet.currencyInfo
     const name = getWalletName(wallet)
 
     const contractAddress = token?.networkLocation?.contractAddress ?? ''
 
-    return (
-      normalizeForSearch(currencyCode).includes(target) ||
-      normalizeForSearch(displayName).includes(target) ||
-      normalizeForSearch(name).includes(target) ||
-      normalizeForSearch(contractAddress).includes(target)
-    )
+    // All search terms must match at least one field (AND logic)
+    return searchTerms.every(term => {
+      return (
+        normalizeForSearch(currencyCode).includes(term) ||
+        normalizeForSearch(displayName).includes(term) ||
+        normalizeForSearch(chainDisplayName).includes(term) ||
+        normalizeForSearch(name).includes(term) ||
+        normalizeForSearch(contractAddress).includes(term)
```

Comment:
So this will do `normalizeForSearch(currencyCode)` repeatedly for each search term. If the user types 3 terms, we'd `normalizeForSearch(currencyCode)` 3 times. Not catastrophic, but we could do `const normalCurrencyCode = normalizeForSearch(currencyCode)` outside the `.every`, so it only happens once.